testlapack/dgehrd.go

```diff
@@ -0,0 +1,204 @@
+// Copyright ©2016 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package testlapack
+
+import (
+	"fmt"
+	"math"
+	"math/rand"
+	"testing"
+
+	"github.com/gonum/blas"
+	"github.com/gonum/blas/blas64"
+)
+
+type Dgehrder interface {
+	Dgehrd(n, ilo, ihi int, a []float64, lda int, tau, work []float64, lwork int)
+}
+
+func DgehrdTest(t *testing.T, impl Dgehrder) {
+	rnd := rand.New(rand.NewSource(1))
+
+	// Randomized tests for small matrix sizes that will most likely
+	// use the unblocked algorithm.
+	for _, n := range []int{2, 3, 4, 5, 10, 34} {
```

Comment:
OK, so what do we want to do?

It seems to me that the most general way of dealing with this is to check for zero length of any numeric slice type and only take the address if its length >= 0.

So `genlapack.pl` would generate for a parameter p of type []T:

```
...
var _p *T
if len(p) > 0 {
    _p = &p[0]
}
return isZero(C.LAPACKE_func(..., (*C.lapack_T)(_p), ...)
...
```